code/game/objects/structures/barricade/deployable.dm

```diff
@@ -26,16 +26,9 @@
 /obj/structure/barricade/deployable/attackby(obj/item/item, mob/user)
 
 	if(iswelder(item))
-		if(!HAS_TRAIT(item, TRAIT_TOOL_BLOWTORCH))
-			to_chat(user, SPAN_WARNING("You need a stronger blowtorch!"))
-			return
-		if(user.action_busy)
-			return
-		var/obj/item/tool/weldingtool/welder = item
-		if(health == maxhealth)
-			to_chat(user, SPAN_WARNING("[src] doesn't need repairs."))
-			return
+		attackby_welder(item, user)
 
+		var/obj/item/tool/weldingtool/welder = item
```

Comment:
Casting here is unnecessary as we don't operate on welder after this.

Reply:
it's required for the proc bellow i think?

Reply:
It's not you can just pass item.